llarp/net/net.cpp

```diff
@@ -469,49 +495,33 @@ namespace llarp
               IPRange{net::ExpandV4(xntohl(ifaddr)), net::ExpandV4(xntohl(ifmask))});
       }
     });
-    // try 10.x.0.0/16
-    byte_t oct = 0;
-    while (oct < 255)
-    {
-      const huint32_t loaddr = ipaddr_ipv4_bits(10, oct, 0, 1);
-      const huint32_t hiaddr = ipaddr_ipv4_bits(10, oct, 255, 255);
-      bool hit = false;
-      for (const auto& range : currentRanges)
+    auto ownsRange = [&currentRanges](IPRange range) -> bool {
+      for (const auto& ownRange : currentRanges)
       {
-        hit = hit || range.ContainsV4(loaddr) || range.ContainsV4(hiaddr);
+        if (ownRange.Contains(range))
+          return true;
       }
-      if (!hit)
-        return loaddr.ToString() + "/16";
-      ++oct;
+      return false;
+    };
+    // generate possible ranges to in order of attempts
+    std::list<IPRange> possibleRanges;
+    for (byte_t oct = 0; oct < 255; ++oct)
+    {
+      possibleRanges.emplace_back(IPRange::FromIPv4(10, oct, 0, 1, 16));
     }
-    // try 192.168.x.0/24
-    oct = 0;
-    while (oct < 255)
+    for (byte_t oct = 0; oct < 255; ++oct)
     {
-      const huint32_t loaddr = ipaddr_ipv4_bits(192, 168, oct, 1);
-      const huint32_t hiaddr = ipaddr_ipv4_bits(192, 168, oct, 255);
-      bool hit = false;
-      for (const auto& range : currentRanges)
-      {
-        hit = hit || range.ContainsV4(loaddr) || range.ContainsV4(hiaddr);
-      }
-      if (!hit)
-        return loaddr.ToString() + "/24";
+      possibleRanges.emplace_back(IPRange::FromIPv4(192, 168, oct, 1, 24));
     }
-    // try 172.16.x.0/24
-    oct = 0;
-    while (oct < 255)
+    for (byte_t oct = 0; oct < 255; ++oct)
     {
-      const huint32_t loaddr = ipaddr_ipv4_bits(172, 16, oct, 1);
-      const huint32_t hiaddr = ipaddr_ipv4_bits(172, 16, oct, 255);
-      bool hit = false;
-      for (const auto& range : currentRanges)
-      {
-        hit = hit || range.ContainsV4(loaddr) || range.ContainsV4(hiaddr);
-      }
-      if (!hit)
-        return loaddr.ToString() + "/24";
-      ++oct;
+      possibleRanges.emplace_back(IPRange::FromIPv4(172, 16, oct, 1, 24));
```

Comment:
The private range here is quite a bit bigger - 172.16.x.x through 172.32.x.x.  This might be more usefully used for /16's:
```C++
    for (byte_t oct = 16; oct < 32; ++oct)
    {
      possibleRanges.emplace_back(IPRange::FromIPv4(172, oct, 0, 1, 16));
    }
```

Reply:
Also we should put this above 192.168.x.0/24, and possibly even before 10.x.0.0/16 checks (since 10.x.x.x is pretty commonly used, and thus more likely to run into conflicts).